### DISCUSSION TITLE
Do we have a global $fetch() helper ?

### DISCUSSION THREAD

**Discussion by kissu** [nuxt/nuxt#9277] [Questions (Nuxt 2)] [answered]
Here is my UI

![image](https://user-images.githubusercontent.com/5133074/117741677-49ef5600-b203-11eb-88a1-6ed819aad58e.png)

Basically, I do have a sidebar (green zone) with various page that I can go on. The red zone is the nested page.
And the orange zone is a dropdown that will toggle every content of data of the page.

The thing is that each page can vary depending of the clicked path on the sidebar and each do have their own `fetch()` hook calls. Moreover, a nested page can have several components, with their own `fetch()` too.

Meanwhile, `this.$fetch()` makes another call on the actual `.vue` file's `fetch()` hook only.

So, a dirty solution would be to:
- take the `<nuxt>` component
- loop on it's children thanks to `.$children`
- for each children, check if he has a `fetch` key
- if it's the case, call `fetch()` on it

It looks like really hacky, not consistent enough and probably not flexible enough.
Don't we have a global `this.fetchGlobal()` method?

**Comment by kissu** [comment, marked as answer]
While writting the end of this question, I've asked myself
> Isn't there any other way to refresh all the data without reloading the whole Nuxt app? There should be a way...

Then, I realized that there was some kind of helper named like this. And...there is actually exactly what I was looking for: `this.$nuxt.refresh()`.

This Nuxt helper will re-fetch every data (`asyncData` or `fetch`) without a subpar hard refresh. More info available here: https://nuxtjs.org/docs/2.x/concepts/context-helpers#refreshing-page-data

So...thanks for this framework ! 🙏🏼 ❤️ 
PS: besides the good vibes, I found it interesting to share my use-case and provide a simple answer to my own question.

**Reply by atinux** [reply]
Thanks for answering your question and guiding new users at the same time ☺️
